pystruct/tests/test_learners/test_crammer_singer_svm.py

```diff
@@ -188,14 +188,13 @@ def test_class_weights():
 def test_class_weights_rescale_C():
     # check that our crammer-singer implementation with class weights and
     # rescale_C=True is the same as LinearSVC's c-s class_weight implementation
-    raise SkipTest("class weight test needs update")
     from sklearn.svm import LinearSVC
     X, Y = make_blobs(n_samples=210, centers=3, random_state=1, cluster_std=3,
                       shuffle=False)
     X = np.hstack([X, np.ones((X.shape[0], 1))])
-    X, Y = X[:170], Y[:170]
 
-    weights = len(Y) / (np.bincount(Y) * len(np.unique(Y)))
+    weights = 1. / np.bincount(Y)
+    weights *= len(weights) / np.sum(weights)
```

Comment:
I think I fixed this in master since then. The formula changed.

Reply:
I'm not clear on those weights. 
Which formula do you refer to?
BTW 1: np.bincount return an array with a 0 => division by zero
BTW 2: I never managed to use weights in pystruct to cope with unbalanced labels. Uniform weighting generally is best. :-O